zcncore/transaction_query_base.go

```diff
@@ -148,49 +148,55 @@ func (tq *TransactionQuery) checkHealth(ctx context.Context, host string) error
 			return ErrNoOnlineSharders
 		}
 	}
-
 	return nil
 }
 
-// randOne random one health sharder
-func (tq *TransactionQuery) randOne(ctx context.Context) (string, error) {
+// getRandomSharder returns a random healthy sharder
+func (tq *TransactionQuery) getRandomSharder(ctx context.Context) (string, error) {
+	ctx, cancel := context.WithCancel(ctx)
+	defer cancel()
+	var mu sync.Mutex
+	done := false
+	errCh := make(chan error, len(tq.sharders))
+	successCh := make(chan string)
+
+	for _, sharder := range tq.sharders {
+		go func(sharder string) {
```

Comment:
This would send requests to all sharders to get one healthy sharder. We should minimize the requests number when possible, so please limit the requests. We can send to 3 sharders concurrently at most, and if all failed, then try next 3 sharders.

Reply:
Okay, but I am wondering if just sending requests to 3 sharders concurrently would be any better than doing it sequentially?

Reply:
Imagine if the selected 3 sharders were offline, but all other sharders are available. Anyway, what I suggested is to send requests to 3 sharders concurrently, if any one  success, then we return. If all requests failed, then we try next 3 sharders and repeat until find one healthy sharder or return error if no sharders alive.

Reply:
Got it, let me make those changes

Reply:
Done